src/OrchardCore.Modules/OrchardCore.Media/TagHelpers/ImageTagHelper.cs

```diff
@@ -44,17 +49,20 @@ IFileVersionProvider fileVersionProvider
 
         public override void Process(TagHelperContext context, TagHelperOutput output)
         {
-            if (string.IsNullOrEmpty(AssetSrc))
+            if (String.IsNullOrEmpty(AssetSrc))
             {
                 return;
             }
 
-            var resolvedSrc = _mediaFileStore != null ? _mediaFileStore.MapPathToPublicUrl(AssetSrc) : AssetSrc;
-            output.Attributes.SetAttribute("src", resolvedSrc);
+            var resolvedUrl = _mediaFileStore != null ? _mediaFileStore.MapPathToPublicUrl(AssetSrc) : AssetSrc;
 
             if (AppendVersion && _fileVersionProvider != null)
             {
-                output.Attributes.SetAttribute("src", _fileVersionProvider.AddFileVersionToPath(_httpContextAccessor.HttpContext.Request.PathBase, resolvedSrc));
+                output.Attributes.SetAttribute(AssetSrcAttributeName, _fileVersionProvider.AddFileVersionToPath(_httpContextAccessor.HttpContext.Request.PathBase, resolvedUrl));
```

Comment:
I think this might need to be `SourceAttributeName` and in the method below as well.

Reply:
Good catch! Fixed.